pkg/artifactory/resource/security/resource_artifactory_scoped_token.go

```diff
@@ -0,0 +1,288 @@
+package security
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/go-resty/resty/v2"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/jfrog/terraform-provider-shared/util"
+	"github.com/jfrog/terraform-provider-shared/validator"
+)
+
+type AccessTokenPostRequest struct {
+	GrantType   string `json:"grant_type"`
+	Username    string `json:"username,omitempty"`
+	Scope       string `json:"scope,omitempty"`
+	ExpiresIn   int    `json:"expires_in,omitempty"`
+	Refreshable bool   `json:"refreshable"`
+	Description string `json:"description"`
+	Audience    string `json:"audience,omitempty"`
+}
+
+type AccessTokenPostResponse struct {
+	TokenId     string `json:"token_id"`
+	AccessToken string `json:"access_token"`
+	ExpiresIn   int    `json:"expires_in"`
+	Scope       string `json:"scope"`
+	TokenType   string `json:"token_type"`
+}
+
+type AccessTokenGet struct {
+	TokenId     string `json:"token_id"`
+	Subject     string `json:"subject"`
+	Expiry      int    `json:"expiry"`
+	IssuedAt    int    `json:"issued_at"`
+	Issuer      string `json:"issuer"`
+	Description string `json:"description"`
+	Refreshable bool   `json:"refreshable"`
+}
+
+func (a AccessTokenPostResponse) Id() string {
+	return a.TokenId
+}
+
+const accessTokensUrl = "access/api/v1/tokens"
+const accessTokenUrl = accessTokensUrl + "/{id}"
+
+func ResourceArtifactoryScopedToken() *schema.Resource {
+
+	const scopesRegex = `^(applied-permissions/user|applied-permissions/admin|applied-permissions/groups|system:metrics:r|system:livelogs:r|artifact:.+:([rwdam\*]|([rwdam]+(,[rwdam]+))))$`
+
+	var scopedTokenSchema = map[string]*schema.Schema{
+		"username": {
+			Type:             schema.TypeString,
+			Optional:         true,
+			ForceNew:         true,
+			ValidateDiagFunc: validation.ToDiagFunc(validation.StringLenBetween(1, 255)),
+			Description:      "The user name for which this token is created. The username is based on the authenticated user - either from the user of the authenticated token or based on the username (if basic auth was used). The username is then used to set the subject of the token: <service-id>/users/<username>. Limited to 255 characters.",
+		},
+		"scopes": {
+			Type:     schema.TypeSet,
+			Optional: true,
+			ForceNew: true,
+			Computed: true,
+			Elem: &schema.Schema{
+				Type:             schema.TypeString,
+				ValidateDiagFunc: validation.ToDiagFunc(
+					validation.All(
+						validation.StringIsNotEmpty,
+						validation.StringMatch(regexp.MustCompile(scopesRegex), "must be one of 'applied-permissions/user', 'applied-permissions/admin', 'applied-permissions/groups', 'system:metrics:r', 'system:livelogs:r', or '<resource-type>:<target>[/<sub-resource>]:<actions>'"),
```

Comment:
Multiline this. Also, since you have specific cases, you should have a validator for just those and then a regex instead of a long regex

```go
validation.Any(
   validation.StringInSlice(...),
   validation.StringMatch()
)
```